ftw/publisher/sender/workflows/example.py

```diff
@@ -47,18 +45,12 @@ def retract_children_first(self):
     @message(_('The referenced object ${item} is not yet published.'))
     @warning_on(interfaces.PUBLISH, interfaces.SUBMIT)
     def references_should_be_published(self):
-        def does_not_share_workflow_with_current_context(target):
-            target_state = getMultiAdapter((target, self.request), IPublisherContextState)
-            if not target_state.has_workflow() and \
-               target_state.get_closest_parent_having_workflow() == self.context:
-                # The reference target is an object within the current object (self.context)
-                # and the target is a direct or indirect child of the current object without
-                # an own workflow.
-                # This means that the reference will most likely be published along with the
-                # current object. In this situation we do not want to warn.
-                return False
-            return True
-        return list(filter(does_not_share_workflow_with_current_context,
+        main_obj = get_main_obj_belonging_to(self.context)
+
+        def reference_is_reference_to_self(target):
```

Comment:
```suggestion
        def reference_is_not_reference_to_self(target):
```
I think the function does the inverse of what the name tells us 😉

Reply:
Oops, you're right 🤦‍♂